src/contract.ts

```diff
@@ -111,25 +113,35 @@ export class AbstractContract {
 
     let error = `VerifyError: ${bsi.errstr}`;
 
-    const lastStepfExec = steps[stepCounter - 1].fExec;
 
-    if(!lastStepfExec) {
-      stepCounter = AbstractContract.findLastfExec(steps, stepCounter);
-    }
 
     // some time there is no opcodes, such as when sourcemap flag is closeed. 
     if(opcodes) {
       const offset = unlockingScriptASM.trim().split(' ').length;
       // the complete script may have op_return and data, but compiled output does not have it. So we need to make sure the index is in boundary.
-      const pc = Math.min(stepCounter -  offset,  opcodes.length -1);
-  
+
+      const lastStepIndex = AbstractContract.findLastfExec(steps, stepCounter);
+
+      if(this._dataPart !== undefined && this._dataPart !== null) {
```

Comment:
simplify
`if(this._dataPart != null) {`

Reply:
done